src/include/wiredtiger.in

```diff
@@ -4773,66 +4748,56 @@ struct __wt_location_handle {
-	 */
-	int (*ss_location_handle)(WT_STORAGE_SOURCE *storage_source,
-	    WT_SESSION *session, const char *location, WT_LOCATION_HANDLE **location_handle);
-
-	/*!
-	 * Return a list of object names for the given location.
-	 *
-	 * @errors
-	 *
-	 * @param storage_source the WT_STORAGE_SOURCE
-	 * @param session the current WiredTiger session
-	 * @param location_handle the location to list
-	 * @param prefix if not NULL, only files with names matching the prefix
-	 *    are returned
-	 * @param limit if not 0, limits the number of objects listed to this number.
-	 * @param[out] object_list the method returns an allocated array of
-	 *    individually allocated strings, one for each object in the location.
-	 * @param[out] countp the number of entries returned
-	 */
-	int (*ss_location_list)(WT_STORAGE_SOURCE *storage_source,
-	    WT_SESSION *session, WT_LOCATION_HANDLE *location_handle, const char *prefix,
-	    uint32_t limit, char ***object_list, uint32_t *countp);
-
-	/*!
-	 * Free memory allocated by WT_STORAGE_SOURCE::location_list.
+	 * Create a customized file system to access the storage source
+	 * objects.
+	 *
+	 * The file system returned behaves as if objects in the specified buckets are
+	 * files in the file system.  In particular, the fs_open_file method requires
+	 * its flags argument to include either WT_FS_OPEN_CREATE or WT_FS_OPEN_READONLY.
+	 * Objects being created are not deemed to "exist" and be visible to
+	 * WT_FILE_SYSTEM::fs_exist and other file system methods until the new handle has
+	 * been closed.  Objects once created are immutable. That is, only objects that
+	 * do not already exist can be opened with the create flag, and objects that
+	 * already exist can only be opened with the readonly flag.  Only objects that
+	 * exist can be transferred to the underlying shared object storage.  This can
+	 * happen at any time after an object is created, and can be forced to happen using
+	 * WT_STORAGE_SOURCE::ss_flush.
+	 *
+	 * Additionally file handles returned by the file system behave as file handles to a
+	 * local file.  For example, WT_FILE_HANDLE::fh_sync synchronizes writes to the
+	 * local file, and does not imply any transferring of data to the shared object store.
+	 *
+	 * The directory argument to the WT_FILE_SYSTEM::fs_directory_list method is normally
+	 * the empty string as the cloud equivalent (bucket) has already been given when
+	 * customizing the file system.  If specified, the directory path is interpreted
+	 * as another prefix, which is removed from the results.
+	 *
+	 * Names used by the file system methods are generally flat.  However, in some
+	 * implementations of a file system returned by a storage source, "..", ".", "/"
+	 * may have a particular meaning, as in a POSIX file system.  We suggest that
+	 * these constructs be avoided when a caller chooses file names within the returned
+	 * file system, they may be rejected by the implementation.  Within a bucket name,
```

Comment:
"they may be rejected" should start a new sentence. (Or maybe a semicolon if you want a closer link)